api/validators/UserSocialMediaUrlControllerRequests.ts

```diff
@@ -0,0 +1,32 @@
+import { Type } from 'class-transformer';
+import { IsDefined, IsNotEmpty, ValidateNested } from 'class-validator';
+import { IsValidSocialMediaUrlType } from '../decorators/Validators';
+import {
+  InsertUserSocialMediaUrlRequest as IInsertUserSocialMediaUrlRequest,
+  UpdateUserSocialMediaUrlRequest as IUpdateUserSocialMediaUrlRequest,
+  SocialMediaUrl as ISocialMediaUrl,
+  SocialMediaType,
+} from '../../types';
+
+export class SocialMediaUrl implements ISocialMediaUrl {
+  @IsDefined()
+  @IsValidSocialMediaUrlType()
+  socialMediaType: SocialMediaType;
+
```

Comment:
nit: we can rename this field across the whole PR to just 'type', since it'd always be in the context of social media (e.g. either in the database table or in an object)

Reply:
nit: let's rename this object to `SocialMedia`